core_lib/src/util/pointerevent.cpp

```diff
@@ -1,34 +1,27 @@
 #include "pointerevent.h"
 
-PointerEvent::PointerEvent(QMouseEvent* event)
+PointerEvent::PointerEvent(QMouseEvent* event, const QPointF& canvasPos)
 {
     mMouseEvent = event;
+    mCanvasPos = canvasPos;
 }
 
-PointerEvent::PointerEvent(QTabletEvent* event)
+PointerEvent::PointerEvent(QTabletEvent* event, const QPointF& canvasPos)
 {
     mTabletEvent = event;
+    mCanvasPos = canvasPos;
 }
 
 PointerEvent::~PointerEvent()
 {
 }
 
-QPoint PointerEvent::pos() const
+QPointF PointerEvent::canvasPos() const
 {
-    if (mMouseEvent)
-    {
-        return mMouseEvent->pos();
-    }
-    else if (mTabletEvent)
-    {
-        return mTabletEvent->pos();
-    }
-    Q_ASSERT(false);
-    return QPoint();
+    return mCanvasPos;
```

Comment:
I think I understand why you did this, to avoid coupling PointerEvent to ViewManager right? I would prefer if it wasn't up to the init method to decide how this value is interpreted but i'm also not sure how to solve it without creating that coupling.

My issue with this right now is that one could be deceived to think that canvasPos is always the canvas position mapped based on ViewManager but in reality it could be something else, depending on the next time someone initializes a PointerEvent.

Reply:
Fair enough, we'll leave it as is 👍